File: frontend/src/types/doubt.ts
export type DoubtMessage = {
  id: number
  doubt_thread_id: number
  role: 'user' | 'assistant'
  content: string
  created_at: string
}

export type DoubtThread = {
  id: number
  parent_message_id: number
  selected_text: string
  created_at: string
  messages: DoubtMessage[]
}

export type DoubtThreadSummary = Omit<DoubtThread, 'messages'>

export type DoubtMessageResponse = {
  user_message: DoubtMessage
  assistant_message: DoubtMessage
}

File: backend/app/schemas/doubt.py
from datetime import datetime

from pydantic import ConfigDict, Field
from sqlmodel import SQLModel


class DoubtThreadCreate(SQLModel):
    selected_text: str = Field(min_length=1)


class DoubtMessageCreate(SQLModel):
    content: str = Field(min_length=1)


class DoubtMessageRead(SQLModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    doubt_thread_id: int
    role: str
    content: str
    created_at: datetime


class DoubtThreadSummary(SQLModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    parent_message_id: int
    selected_text: str
    created_at: datetime


class DoubtThreadRead(DoubtThreadSummary):
    messages: list[DoubtMessageRead] = []


class DoubtMessageResponse(SQLModel):
    user_message: DoubtMessageRead
    assistant_message: DoubtMessageRead
